pint_server/app.py

```diff
@@ -425,11 +439,35 @@ def get_provider_servers(provider):
             for server in servers]
 
 
+def trim_images_payload(images):
+    payload_size = jsonify(images=images).content_length
+
+    if payload_size >  MAX_PAYLOAD_SIZE:
+        # NOTE: assuming the size of the entries are evenly distributed, we
+        # determine the percentage of entries to trim from the end of the list,
+        # as the list is sorted in decending order by publishedon date, by
+        # calculating the percentage over the maximum payload size. Then
+        # trim the same percentage off the list, rounding up to be safe.
+        trim_size  = math.ceil(
+            ((payload_size - MAX_PAYLOAD_SIZE) / payload_size) * len(images))
+        images = images[:-trim_size]
+
+        # Now make sure we don't have partial data by finished triming all the
+        # images from all regions that have the same publishedon date that of
+        # the last image that got trimmed.
+        publishedon = images[-1]['publishedon']
+        while images[-1]['publishedon'] == publishedon:
```

Comment:
Minor performance nit...

This while loop is going to lead to multiple `images` list modification operations because we pop off each extra entry individually, once per iteration.

If you instead just decrement the index value on each iteration, then do a single slice operation at the end to determine the final trimmed images list, that is likely to be slightly more efficient, performance wise.

Of course this is probably very minor in the overall scheme of things. ;-)

Similarly if there is a concern about avoiding too many list deref's, you could combine a computed index value with a step factor, e.g. go back in decrements of X (e.g. number of provider regions) and then use a binary search to find the first entry with the new publishedon value.

Again the performance gain would probably be negligible, unless we could potentially be dealing with many hundreds of images being published on the same day...

Reply:
I agree on the pop... would be better to find the index and then make a single additional cut... I worry the binary search, which performant, may be overly complicated for the number of rows we're talking about... personally I'd probably keep it simple if possible.

Reply:
Yeah I agree on binary search... Not sure we are anywhere near having 1000s of entries with same publishedon date...

Reply:
no need for binary search as we don't have 1000s of entries to trim. We need to avoid binary search as performance is terrible.